x/campaign/simulation/store.go

```diff
@@ -179,7 +180,7 @@ func GetAccountWithVouchers(
 		coinCampID, err := types.VoucherCampaign(coin.Denom)
 		if err == nil && coinCampID == campID {
 			// retain a random portion of the balance in the range [0, coin.Amount)
-			retainAmt := sdk.NewInt(rand.Int63n(coin.Amount.Int64()))
+			retainAmt := sdk.NewInt(r.Int63n(coin.Amount.Int64()))
```

Comment:
I would consider using https://github.com/cosmos/cosmos-sdk/blob/master/types/simulation/rand_util.go#L43 instead

Reply:
or better probably `RandomAmount()`

Reply:
Question: is there a specific reason we calculate `retainAmt` and subtract coin's amount from it, instead of directly set a random amount in `coins` structure?

Reply:
I think its because this is just a bit simpler than building a new `coin` with the same `denom` as the `coin` we are operating on

Reply:
I changed the logic a bit to avoid 0 token that trigger errors since `RandPositiveInt` include the max amount in the range

Reply:
nope, what you did seems perfect